src/integrationTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/client/payhub/PayHubClientTest.java

```diff
@@ -0,0 +1,147 @@
+package uk.gov.hmcts.reform.bulkscan.payment.processor.client.payhub;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.tomakehurst.wiremock.client.ResponseDefinitionBuilder;
+import com.google.common.collect.ImmutableList;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.client.payhub.request.PaymentRequest;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.client.payhub.response.PaymentResult;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.config.IntegrationTest;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.badRequest;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalToJson;
+import static com.github.tomakehurst.wiremock.client.WireMock.okJson;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.serverError;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.google.common.net.HttpHeaders.CONTENT_TYPE;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR;
+
+
+@IntegrationTest
+public class PayHubClientTest {
+
+    @Autowired
+    private PayHubClient client;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @Test
+    public void should_return_Ok_when_everything_is_ok_with_request() throws JsonProcessingException {
```

Comment:
I think having both 200 and 201 response code doesn't make sense. I have emailed F&P team to confirm. We may only have 201 for create case scenario.

Reply:
but if they return any 2xx status code we will not retry?

Reply:
201 means payment record for the envelope have been created not sure why we want to retry  if it is 201 ?

Reply:
we do not care what 2xx code they return ? 
we will not retry if they return  any 2xx like 200 or 201,202,204 ?

Reply:
ok can do that way